Source/UI/Home/HomeView.swift

```diff
@@ -160,51 +116,31 @@ struct HomeView: View, HelpDrawerHost {
                 }
             }
         }
-        .alert(
-            Localized.error.text,
-            isPresented: shouldShowAlert,
-            actions: {
-                Button(Localized.tryAgain.text) {
-                    Task {
-                        await loadFromScratch()
-                    }
-                }
-                Button(Localized.cancel.text, role: .cancel) {
-                    shouldShowAlert.wrappedValue = false
-                }
-            },
-            message: {
-                Text(errorMessage ?? "")
-            }
-        )
-        .task {
-            if messages == nil {
-                await loadFromScratch()
-            }
-        }
-        .refreshable {
-            await loadFromScratch()
-        }
         .onReceive(NotificationCenter.default.publisher(for: .didChangeHomeFeedAlgorithm)) { _ in
-            Task.detached {
-                await loadFromScratch()
+            Task {
+                await dataSource.loadFromScratch()
             }
         }
         .onReceive(NotificationCenter.default.publisher(for: .didUpdateRelationship)) { _ in
-            Task.detached {
-                await loadFromScratch()
+            Task {
+                await dataSource.loadFromScratch()
             }
         }
         .onReceive(NotificationCenter.default.publisher(for: .didPublishPost)) { _ in
-            Task.detached {
-                await loadFromScratch()
+            Task {
+                await dataSource.loadFromScratch()
             }
         }
         .onReceive(NotificationCenter.default.publisher(for: .didRefresh)) { _ in
             Task.detached {
```

Comment:
Is there a reason this Task is still detached but the others aren't?

Reply:
I added `@MainActor` to `checkNewItemsIfNeeded` and removed the `Task.detached` and made `loadMore` async.